lightning/src/ln/channelmanager.rs

```diff
@@ -4260,11 +4319,32 @@ where
 		if let None = peer_state_mutex_opt {
 			return Err(MsgHandleErrInternal::send_err_msg_no_close(format!("Can't find a peer matching the passed counterparty node_id {}", counterparty_node_id), msg.temporary_channel_id.clone()))
 		}
+
+		// Get the number of peers with channels, but without funded ones. We don't care too much
+		// about peers that never open a channel, so we filter by peers that have at least one
+		// channel, and then limit the number of those with unfunded channels.
+		let peers_without_funded_channels = self.peers_without_funded_channels(|node| !node.channel_by_id.is_empty());
```

Comment:
Note I think  the comment and the variable `peers_without_funded_channels` actually apply to the second check. I don't think there is a comment for the max-unfunded-chans-per-peers, something like "Get the number of unfunded channel for this peer and if this beyond the limit of unfunded chans per peers, reject this new channel opening".

Reply:
Yeah better with the swap.